app/va/va_profile/va_profile_client.py

```diff
@@ -189,21 +267,9 @@ def get_telephone_with_permission(
 
         contact_info: ContactInformation = profile.get('contactInformation', {})
 
-        telephones: list[Telephone] = contact_info.get(self.PHONE_BIO_TYPE, [])
-        sorted_telephones = sorted(
-            [phone for phone in telephones if phone['phoneType'] == PhoneNumberType.MOBILE.value],
-            key=lambda phone: iso8601.parse_date(phone['createDate']),
-            reverse=True,
-        )
-        if sorted_telephones:
-            if (
-                sorted_telephones[0].get('countryCode')
-                and sorted_telephones[0].get('areaCode')
-                and sorted_telephones[0].get('phoneNumber')
-            ):
-                self.statsd_client.incr('clients.va-profile.get-telephone.success')
-            telephone_result = f"+{sorted_telephones[0]['countryCode']}{sorted_telephones[0]['areaCode']}{sorted_telephones[0]['phoneNumber']}"
-            return VAProfileResult(telephone_result, communication_allowed, permission_message)
+        telephone = self.get_mobile_telephone_from_contact_info(contact_info)
+        if telephone:
+            return VAProfileResult(telephone, communication_allowed, permission_message)
```

Comment:
Can we flip the logic please?
```suggestion
        if not telephone:
            self.statsd_client.incr('clients.va-profile.get-telephone.failure')
            self._raise_no_contact_info_exception(self.PHONE_BIO_TYPE, va_profile_id, contact_info.get(self.TX_AUDIT_ID))
        return VAProfileResult(telephone, communication_allowed, permission_message)
```
We want to avoid returning other than at the beginning (guard statements are fine) or as the last line. Shouldn't affect results nor tests.

Reply:
@AdamKing0126